src/pages/top-speaker/list/index.tsx

```diff
@@ -141,14 +153,18 @@ const TopSpeakerList: React.FC = () => {
 		$("#topSpeakerViewModal").modal("show");
 	};
 
-	const handleSetUpdateTopSpeaker = (topSpeakerId: string) => {
-		dispatch(setTopSpeakerId(topSpeakerId));
-		$("#topSpeakerUpdateModal").modal("show");
+	const handleSetUpdateTopSpeaker = (topSpeaker: any, topSpeakerId: string) => {
+		if (topSpeaker && (permission === "ROOT_ADMIN" || permission === "ADMIN")) {
+			dispatch(setTopSpeakerId(topSpeakerId));
+			$("#topSpeakerUpdateModal").modal("show");
+		}
```

Comment:
Add an `else` part with some warning message. Otherwise, user won't know what is happened.
```
if (permission === "ROOT_ADMIN" || permission === "ADMIN" || permission === "EDITOR") {
   dispatch(setApplicationId(applicationId));
   $("#applicationDeleteModal").modal("show");
} else {
   // Send a error or warning notification
}
```